packages/perseus/src/widgets/interactive-graphs/graphs/components/angle.tsx

```diff
@@ -0,0 +1,173 @@
+    const endAngle = Math.atan2(endY - centerY, endX - centerX);
+    // const normalizedAngle = angle > 2 * Math.PI ? angle - 2 * Math.PI : angle;
+    const x1 = centerX + radius * Math.cos(startAngle);
+    const y1 = centerY + radius * Math.sin(startAngle);
+    const x2 = centerX + radius * Math.cos(endAngle);
+    const y2 = centerY + radius * Math.sin(endAngle);
+
+    // Fourth point that would create a rhomboid
+    // Used to calculate the inside of the angle and to create square
+    const [x3, y3] = vec.add(
+        centerPoint,
+        vec.add(vec.sub([x1, y1], centerPoint), vec.sub([x2, y2], centerPoint)),
+    );
+
+    // Midpoint betwen ends of arc
+    const isInside = shouldDrawArcInside(
+        [x3, y3],
+        centerPoint,
+        range,
+        polygonPoints,
+    );
+
+    const largeArcFlag = isInside ? 1 : 0;
+    const sweepFlag = isInside ? 1 : 0;
+
+    const arc = `M ${x1} ${y1} A ${radius} ${radius} 0 ${largeArcFlag} ${sweepFlag} ${x2} ${y2}`;
+    const square = `M ${x1} ${y1} L ${x3} ${y3} M ${x3} ${y3} L ${x2} ${y2}`;
+
+    const angleLabelNumber = parseFloat((angle * (180 / Math.PI)).toFixed(1));
+    const angleLabel = Number.isInteger(angleLabelNumber)
+        ? angleLabelNumber
+        : "≈ " + angleLabelNumber;
+
+    return (
+        <>
+            <defs>
+                <filter
+                    id="background"
+                    x="-5%"
+                    width="110%"
+                    y="0%"
+                    height="100%"
+                >
+                    <feFlood floodColor="#FFF" floodOpacity="0.5" />
+                    <feComposite operator="over" in="SourceGraphic" />
+                </filter>
+            </defs>
+            <g
+                style={{
+                    transform: `var(--mafs-view-transform) var(--mafs-user-transform)`,
+                }}
+            >
+                <path
+                    d={isRightAngle(angle) ? square : arc}
+                    strokeWidth={0.02}
+                    fill="none"
+                />
+            </g>
+            <Text
+                x={x3}
+                y={y3}
+                size={15}
+                svgTextProps={{
+                    filter: "url(#background)",
+                    fontWeight: "bold",
+                }}
+                // Shift position if text is too close to movable point
+                attach={y3 - centerY > 0 ? "s" : "n"}
+                attachDistance={
+                    Math.abs(y3 - centerY) < 0.2 ||
+                    vec.dist([x3, y3], centerPoint) < 0.3
+                        ? 20
+                        : 10
+                }
+            >
+                {angleLabel}°
+            </Text>
+        </>
+    );
+};
+
+const isRightAngle = (angle) => {
+    if (angle > Math.PI) {
+        angle = Math.abs(angle - Math.PI);
+    }
+    return Math.abs(angle - Math.PI / 2) < 0.01;
```

Comment:
I think the tolerance on this should be smaller than `0.01`. 0.01 radians is ~0.5 degrees.
<img width="392" alt="Screen Shot 2024-03-22 at 9 33 38 AM" src="https://github.com/Khan/perseus/assets/693920/3c672f6a-b630-428b-8060-d3b3fa9fe690">